src/main/java/org/skriptlang/skript/addon/SkriptAddon.java

```diff
@@ -0,0 +1,61 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.addon;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A Skript addon is an extension to Skript that expands its features.
+ *
+ */
+@ApiStatus.Experimental
+public interface SkriptAddon {
+
+	/**
+	 * @return The name of this addon.
+	 */
+	String name();
+
+	/**
+	 * The path to the directory on disk containing data files for this addon.
+	 *  For example, this may include language files that have been saved to enable user customization.
+	 * @return A string representing the path to the directory.
+	 *  Null if this addon does not store any data files.
+	 */
+	@Nullable
+	default String dataFileDirectory() {
+		return null;
+	}
+
+	/**
+	 * The path to the directory containing language files for this addon.
+	 * When searching for language files on the jar, this will be used as the path.
+	 * When searching for language files on the disk, this will be used along with {@link #dataFileDirectory()}.
+	 * That is, it is expected that the path <code>dataFileDirectory() + languageFileDirectory()</code> would
+	 *  lead to this addon's language files on the disk.
+	 * @return A string representing the path to the directory.
+	 *  Null if this addon does not have any language files.
+	 */
+	@Nullable
+	default String languageFileDirectory() {
+		return null;
+	}
```

Comment:
Do we want to link Skript to the file system by design? Wouldn't it be better if translations were handled by a separate class which can be accessed through the SkriptAddon instance? This new translations class would have capabilities for parsing from a folder or just manually adding them. This way you as the addon developer have full control over what's done, and aren't required to place files somewhere.

There should be some kind of basic "BukkitSkriptAddon" which handles a lot of common things, but that shouldn't mean the top SkriptAddon interface should be aware of that.

Reply:
Maybe we'd want to introduce a LocalizedSkriptAddon interface that contains these methods? I do agree that these methods shouldn't be here on the top SkriptAddon interface. In the future, after a rework of the Language system parsing from sources other than files could be supported and the interface (in terms of its methods) could be re-examined. Separating it out for now would be good though

Reply:
Yeah that sounds good.